scipy/signal/_signaltools.py

```diff
@@ -26,6 +26,11 @@
 from ._fir_filter_design import firwin
 from ._sosfilt import _sosfilt
 
+from scipy._lib._array_api import (
+    array_namespace, is_torch, is_numpy, xp_copy, xp_size
+)
+import scipy._lib.array_api_compat.numpy as np_compat
+from scipy._lib.array_api_extra import atleast_nd
```

Comment:
non-blocking at all but the style in the examples of the array-api-extra docs is always:

```python
import array_api_extra as xpx
...
xpx.atleast_nd(...)
```

I think it is nice to keep the style more consistent with the calls to `xp.` functions. But of course this is just a preference.